gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,398 @@
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  @Override
+  public boolean dropNamespace(Namespace namespace) {
+    try {
+      if (namespace.levels().length == 0) {
+        // Used by `DROP NAMESPACE <catalog>`. Deletes the BLMS catalog linked by Iceberg catalog.
+        client.deleteCatalog(catalogName);
+        LOG.info("Deleted BigLake catalog: {}", catalogName.toString());
+      } else if (namespace.levels().length == 1) {
+        client.deleteDatabase(getDatabaseName(namespace));
+        // We don't delete the data file folder for safety. It aligns with HMS's default behavior.
+        // We can support database or catalog level config controlling file deletion in future.
+      } else {
+        throw new IllegalArgumentException(invalidNamespaceMessage(namespace));
+      }
+    } catch (NoSuchNamespaceException e) {
+      LOG.warn("Dropping namespace failed", e);
+      return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) {
+    Database.Builder builder;
+    try {
+      builder = getDatabase(namespace).toBuilder();
+    } catch (IllegalArgumentException e) {
+      LOG.warn(
+          "setProperties is only supported for tables and databases, namespace {} is not supported",
+          namespace.levels().length == 0 ? "empty" : namespace.toString(),
+          e);
+      return false;
+    }
+
+    HiveDatabaseOptions.Builder optionsBuilder = builder.getHiveOptionsBuilder();
+    properties.forEach(optionsBuilder::putParameters);
+    client.updateDatabaseParameters(getDatabaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) {
+    Database.Builder builder;
+    try {
+      builder = getDatabase(namespace).toBuilder();
+    } catch (IllegalArgumentException e) {
+      LOG.warn(
+          "removeProperties is only supported for tables and databases, namespace {} is not"
+              + " supported",
+          namespace.levels().length == 0 ? "empty" : namespace.toString(),
+          e);
+      return false;
+    }
+
+    HiveDatabaseOptions.Builder optionsBuilder = builder.getHiveOptionsBuilder();
+    properties.forEach(optionsBuilder::removeParameters);
+    client.updateDatabaseParameters(getDatabaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.levels().length == 0) {
+      // Calls getCatalog to check existence. BLMS catalog has no metadata today.
+      client.getCatalog(catalogName);
+      return new HashMap<String, String>();
+    } else if (namespace.levels().length == 1) {
+      return getMetadata(getDatabase(namespace));
+    } else {
+      throw new IllegalArgumentException(invalidNamespaceMessage(namespace));
```

Comment:
I think this should fail with `NoSuchNamespaceException`.

Reply:
Done.